vice.go

```diff
@@ -61,6 +63,10 @@ const (
 	// Canceled means the operation has been canceled, typically by the
 	// consumer.
 	Canceled
+
+	// Precondition indicates that the state of the system does not meet the
+	// requirements to fulfull the request.
+	Precondition
```

Comment:
Can you give an example here? It doesn't need to be in the comments though it could be. I just can't imagine quite what i'd use it for?